src/app/dataset-view/dataset.component.ts

```diff
@@ -281,8 +283,19 @@ export class DatasetComponent extends BaseDatasetDataComponent implements OnInit
     }
 
     public onRunSQLRequest(params: DatasetRequestBySql): void {
+        const startTime = new Date().valueOf();
+        this.sqlLoading = true;
         this.datasetService
-            .requestDatasetDataSqlRun(params) // TODO: Propagate limit from UI and display when it was reached
+            // TODO: Propagate limit from UI and display when it was reached
+            .requestDatasetDataSqlRun(params)
+            .pipe(
+                finalize(() => {
+                    this.sqlLoading = false;
+                    const endTime = new Date().valueOf();
+                    this.sqlRequestTime = endTime - startTime;
```

Comment:
As I understand, this difference is in milliseconds. We should name the result `sqlRequestTimeInMilliseconds` and propagate it like this, so that the data type is clear.

Reply:
Fixed